ygnmi/types.go

```diff
@@ -170,6 +171,11 @@ func (lq *leafBaseQuery[T]) pathStruct() PathStruct {
 	return lq.ps
 }
 
+// subPaths returns the path struct containing the path for the Query.
+func (lq *leafBaseQuery[T]) subPaths() []PathStruct {
+	return []PathStruct{lq.ps}
```

Comment:
What's the rationale for returning itself? For batch we don't want itself right? Is this to serve as the default and batch can override?

Reply:
a leaf query can never subscribe to paths that differ from the "schema" path. but it still needs to return the path to subscribe to.
a nonleaf query can subscribe to children of the "schema" path

Reply:
I see I missed that this is for leafs only. Do you think it makes sense to add a comment here explaining this?

Reply:
added